### PR TITLE
Upgrade to v4

### PR DESCRIPTION
**Pull request by msv**
Upgrades eslint to require at least 4.10.0. Any of the breaking changes shouldn't effect these rules.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,24 +22,18 @@ __default__
...
 extends: braintree
-rules:
-  locally-overriden-rule: 1
 ```
 
 __browserify__
 ```yaml
 ---
 extends: braintree/client
-rules:
```
Comment (crookedneighbor) [inline README.md:33]: Why are we removing examples for how to override the config? Is this not possible with eslint@v4?

Comment (msv) [reply README.md:33]: @crookedneighbor I can't actually find any documentation on `locally-overriden-rule`, and I get an error when using it that it's an undefined property. To override rules you just add them to your local rules, overriding any options if they were set in a parent as well.

Comment (crookedneighbor) [reply README.md:33]: This is a readme to show how to override rules. `locally-overriden-rule` is not a real rule, just an example for how you would do it with a real rule. 

If that's confusing, we could have a real rule here.

Comment (lilaconlee) [reply README.md:33]: A real rule wouldn't be bad here

Comment (lilaconlee) [reply README.md:33]: or something like `example-locally-overriden-rule`

**Review by crookedneighbor — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -48,8 +42,31 @@ __browserify + es6__
...
-  locally-overriden-rule: 1
 ```
 
 You can specify a `.eslintrc` for a subdirectory to change the rules that are enforced. For instance, in a node project you could extend from `eslint-config-braintree/server` at the top-level, and `eslint-braintree-config/client` at the `public/.eslintrc` level.
+
+See [Configuration File Formats](https://eslint.org/docs/user-guide/configuring#configuration-file-formats) for information on all supported `.eslintrc` file formats.
+
+To override rules, add the new config under `rules` in your rc file. Be sure to properly override any options set by the parent. See [Extending Configuration Files](https://eslint.org/docs/user-guide/configuring#extending-configuration-files) for details.
+
+For exmaple, to change the `no-extra-parens` rule to warn instead of error:
```
Comment (bluk) [inline README.md]: `example` minor misspelling

File: package.json [hunk 3]
```diff
@@ -13,10 +13,10 @@
...
   },
   "peerDependencies": {
-    "eslint": "^2.7.0"
+    "eslint": "^4.10.0"
   },
   "devDependencies": {
-    "eslint": "2.7.0",
-    "mocha": "3.0.0"
+    "eslint": "^4.10.0",
+    "mocha": "^3.0.0"
```
Comment (crookedneighbor) [inline package.json:20]: mocha is actually on v4 now. Could we try updating it?

**Review by lilaconlee — APPROVED** [review]

**Pull request merged**